classes/form/studentattendance.php

```diff
@@ -56,6 +56,9 @@ public function definition() {
                 unset($statuses[$status->id]);
                 $disabledduetotime = true;
             }
+            if ($status->availablebeforesession == 0  && time() < $attforsession->sessdate) {
```

Comment:
Thanks @sumaiyamannan  this looks good - just one thing on this line I think is slightly wrong:

the $status->studentavailability setting allows for status values to be open "before" the official start of the session - for example, the session may start at 1pm, but students may be able to start flagging their attendance as present a few minutes before the session start (if allowed by the teacher.)

so:
```
time() < $attforsession->sessdate
```
should be closer to:
```
time() < ($attforsession->sessdate + ($status->studentavailability * 60))
```

hopefully that makes sense?

Reply:
heh - nope... doesn't make sense... $status->studentavailability isn't the var I was meaning needs to be checked there... its the session "studentsearlyopentime" I really meant needs to be checked - if that setting is set, we need to make sure that the status values are still available for student marking.